### DISCUSSION TITLE
Write two std::vectors to h5 dataset

### DISCUSSION THREAD

**Discussion by mgaimann** [BlueBrain/HighFive#773] [Questions] [answered]
Hi everyone,

just a very naive question: I have two std::vectors `observable` and `observableY` and I would like to dump them to h5.

To obtain a dataset with two rows, I figured that this is the easiest option:
```
std::vector<std::vector<float_type>> observable2d = {observable, observableY};
H5Easy::dump(file, h5path, observable2d);
```

To obtain a dataset with two columns (which I need in my case), this works for me:
```
HighFive::DataSpace dataspace{{observable.size(), 2}};
HighFive::DataSet dataset = file.createDataSet(h5path, dataspace, HighFive::create_datatype<float>());

dataset.select({0, 0}, {observable.size(), 1}).write(observable);
dataset.select({0, 1}, {observable.size(), 1}).write(observableY);
```

I was just wondering whether there is an even easier version for doing this, or whether this is how I am supposed to do it with HighFive, any suggestions are welcome, thank you.

**Comment by 1uc** [comment, marked as answer]
I think your code looks reasonable; and I'm not aware of any short cuts.
